src/utils/memoFn.js

```diff
@@ -0,0 +1,70 @@
+// let cachedResult = 0;
+
+/**
+ * Memoization wrapper for common used PURE helper functions.
+ * The function will be cached only if it receives number, string or boolean as params
+ * PURE means that function accepts at least a single argument and returns
+ * same result all the time same arguments are used.
+ * In order to memoize the function, it should be defined only once as a static function - outside react components.
+ *
+ *
+ * @export
+ * @param {Function} fn - pure function with at least single argument
+ * @returns {Function} - same function with a cache
+ */
+export function memoFn(fn) {
+  const cache = {};
+
+  return (...args) => {
+    // if no arguments used we just want the developer and run the function as is
+    if (args.length === 0) {
+      if (__DEV__) {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `memoized function ${fn.name} was called with no arguments`
+        );
+      }
+
+      // Call it anyway to not break stuff
+      return fn.apply(null, args);
+    }
+
+    // we check for arguments to be number/boolean/string
+    for (let i = 1; i < args.length; i++) {
+      const arg = args[i];
+      if (
+        typeof arg !== 'number' &&
+        typeof arg !== 'boolean' &&
+        typeof arg !== 'string'
```

Comment:
Since this is a new low-level util and the function has such strict type requirements for the arguments, I think this should be written in TypeScript.